### DISCUSSION TITLE
Finding a good layout - Cardiogram Room

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2407] [General Diagnostics]
I am exploring good design layouts for each room, in this post I am focusing on the Cardiogram room.
**The Goal**
Keeping the room as small as possible while keeping the doctor as happy as possible(avoid being unhappy and wanting a pay rise) while optimizing turnaround time for handyman plant watering and patient walk times.

**Version 1.00**
![Screenshot 2023-09-24 101349](https://github.com/CorsixTH/CorsixTH/assets/13173133/48189e3c-b985-4c80-875a-6cd814fb8ba5)
**Items**
1x Cardio
1x Screen
1x Bin
1x Fire extinguisher
1x Plant 
1x Radiator

**Summary**
The most important aspect of the Cardiogram Room is the positing of the Cardio and Screen, in this layout the patient can walk into the room and turn left into the screen then they can walk up onto the Cardio. This has a very optimize walk path as there are no wasted walk turns. They also can pass the bin on the way to the screen. The plant is at the door for fast watering.
The Fire extinguisher (while unable to be in the center of the room, 2 tile radius) is in quite a optimize position, a second one could be added behind the screen to the right of the room. I have used 1 Radiator for this room.
  
Lets start the decision, can this design be improved and if so how?

**Version 1.01**
![Screenshot 2023-09-25 104457](https://github.com/CorsixTH/CorsixTH/assets/13173133/91f59f72-5530-44ca-b33f-8d1efde1813f)
Room is now a little bigger 4x5.
Changed the location to the door and screen.
Plant is still next to door.
Bin is also next to door.
The room should have a good heat map
The Fire extinguisher should mostly still cover the whole room. 
The patient can now just walk into the screen without any turns, change and walk right onto the Cardio.

**Comment by PaveMentman** [comment]
---
As a refresher:
For Cardigram Room,
the patient-routine-path is following:
1. Enter the room
2. Remove clothing on the the North-West / up-left-square of the Screen 
3. Cardio 
4. Put clothes on behind the Screen
5. Exit the room

Important-reminder:
"Screen" cannot be rotated! 

Here is my most likely best optimized Cardio-room when the entrance is place North-West.
( cropped from an archive-image, so semi apologies for messiness ).

In case someone is wondering about my radiator placements:
I tend to put radiators under the outer-wall-windows and next to the building entrances as more of realism-cosmetic.
And then I slight increase the heating from the average.
This is most certainly is not the most min-max-cost-efficient in overall since this game (most likely) does not have draft-element coded in.
But then again, primarily my screenshot(s) are here to give inspirations.

![2023-09-24-15-21-04-948-i_view64-2023-08-28-10-40-46-858-CorsixTH-CorsixTH png_-_Ir](https://github.com/CorsixTH/CorsixTH/assets/6709463/ad02b1c3-c09b-490a-baf5-e1d73c581921)
---

**Reply by ageekhere** [reply]
Updated to Version 1.01, i moved the Cardio to be more inline with the screen so the patient does not need to make a turn while heading to the Cardio. Also made it so the Fire extinguisher can mostly cover the room (2 square radius for happiness).

**Comment by borsitsb** [comment]
I found that patients tend to choose a slightly longer path to the screen when entering the room. Interestingly, they found the direct path when exiting the room. I blocked their way with a fire extinguisher. I'm not sure if the bin is in the right place. What do you think?

<img width="1276" height="812" alt="CleanShot 2025-11-28 at 18 14 44@2x" src="https://github.com/user-attachments/assets/9cce46f2-a5e6-4a0b-8c77-0b1324077418" />

**Reply by ageekhere** [reply]
this is what i had in the end
<img width="1002" height="685" alt="Screenshot 2023-09-25 104457" src="https://github.com/user-attachments/assets/476face0-b910-40de-9ac3-b26405732b4d" />
There is room to add more plants or other items

**Reply by borsitsb** [reply]
4 fire extinguishers maximize the chance that a damaged machine [won't explode](https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Lua/entities/machine.lua), so I always find a way for them. How do you take screenshots with transparent walls? I'm working on a database of optimal layouts for each door direction. It would be nice to capture the hidden details. This is easier than creating a blueprint, like a drawing, for each.

By the way, this is the first time I've played a game while reading its source code to understand how it works under the hood. It's a fun exercise 😃 .

**Reply by ageekhere** [reply]
I use x to enable transparent walls
Here is an improvement 
<img width="779" height="523" alt="Screenshot 2025-12-04 211133" src="https://github.com/user-attachments/assets/8eebc656-ac4e-4d8c-afb3-23b895265b19" />

**Reply by ARGAMX** [reply]
> How do you take screenshots with transparent walls?

Hold `X` key for transparent walls. Press `Shift + X` for permanent transparent walls. 👍

**Reply by ageekhere** [reply]
Another option with 3 heaters
<img width="758" height="512" alt="Screenshot 2025-12-04 211951" src="https://github.com/user-attachments/assets/da073d21-d5d6-428c-b848-ccea733cce38" />

**Reply by borsitsb** [reply]
Does this room require 2-3 heaters? I thought 1 is enough for this size.

**Reply by ARGAMX** [reply]
> Does this room require 2-3 heaters? I thought 1 is enough for this size.

I thought too. 3 is definitely too much.

**Reply by ageekhere** [reply]
It is a good question on how many heaters are needed for a room, with more can you decrease the heat which will save money. Try setting the heat to the lowest value and see if 1, 2 or 3 heaters keep the room warm.
